Rubberduck.RibbonDispatcher/Concrete/RibbonButton.cs

```diff
@@ -0,0 +1,66 @@
+﻿using System;
+using System.Runtime.InteropServices;
+using System.Diagnostics.CodeAnalysis;
+using stdole;
+
+using Rubberduck.RibbonDispatcher.ControlMixins;
+using Rubberduck.RibbonDispatcher.AbstractCOM;
+
+namespace Rubberduck.RibbonDispatcher.Concrete {
+    /// <summary>The ViewModel for Ribbon Button objects.</summary>
+    [SuppressMessage("Microsoft.Interoperability", "CA1409:ComVisibleTypesShouldBeCreatable",
+        Justification = "Publc, Non-Creatable, class with exported Events.")]
+    [Serializable]
+    [CLSCompliant(true)]
+    [ComVisible(true)]
+    [ClassInterface(ClassInterfaceType.None)]
+    [ComSourceInterfaces(typeof(IClickedEvents))]
+    [ComDefaultInterface(typeof(IRibbonButton))]
+    [Guid(RubberduckGuid.RibbonButton)]
+    public class RibbonButton : RibbonCommon, IRibbonButton,
+        ISizeableMixin, IClickableMixin, IImageableMixin {
+        internal RibbonButton(string itemId, IResourceManager mgr, bool visible, bool enabled, RdControlSize size,
+                ImageObject image, bool showImage, bool showLabel) : base(itemId, mgr, visible, enabled) {
+            this.SetSize(size);
+            this.SetImage(image);
+            this.SetShowImage(showImage);
+            this.SetShowLabel(showLabel);
+        }
+
+        #region Publish ISizeableMixin to class default interface
```

Comment:
The regions are indicating far more than comments - they are signals that the code within must not be re-ordered, and must be character-by-character identical with other eponymous regions. They are in fact "inviolable regions", and must be treated as such. Hence my use of regions.

I rarely use regions, as they are typically superficial attempts to make over-large methods and classes appear to be somewhat less so. But these ***Mixin** regions* are identified as such because messing with them is likely to break the code. The only other mechanism to properly signal this, that I have yet come up with, is to make the classes partial and put each Mixin block in a separate file - but that seemed even worse than the use of regions. However, if you prefer that  I can go with it.